packages/shell-api/src/stream-processor.ts

```diff
@@ -57,6 +58,43 @@ export default class StreamProcessor extends ShellApiWithMongoClass {
     });
   }
 
+  /**
+   * modify can be used like so:
+   *  Change the pipeline:
+   *    sp.name.modify(pipeline)
+   *  Change the pipeline with additional options:
+   *    sp.name.modify(pipeline, {resumeFromCheckpoint: false})
+   *  For modify requests that don't change the pipeline:
+   *    sp.name.modify({resumeFromCheckpoint: false})
+   */
+  @returnsPromise
+  async modify(
+    pipelineOrOptions: Document[] | Document,
+    options: Document = {}
+  ) {
+    if (Array.isArray(pipelineOrOptions)) {
+      options['pipeline'] = pipelineOrOptions;
```

Comment:
```suggestion
      options.pipeline = pipelineOrOptions;
```

or, even better:

```suggestion
      options = { ...options, pipeline: pipelineOrOptions };
```

since best practice is not to modify arguments passed to functions

Reply:
Thanks, done.